2-mandatory/3-shopping-list.js

```diff
@@ -19,7 +19,19 @@ let pantry = {
 };
 
 function createShoppingList(recipe) {
-  // write code here
+  const allIngredients = Object.values(pantry).flat(); // flat() join the two array in one
+  const missingItems = recipe.ingredients.filter((ingredient) => {
+    if (allIngredients.includes(ingredient)) {
+      return false; // if has the ingredient , remove it
+    } else {
+      return true; // if has not the ingredient, put it in the object
+    }
+  });
+  const objectReturn = {
+    name: recipe.name,
+    items: missingItems,
+  };
+  return objectReturn;
```

Comment:
Good written function. Some advices on variable naming
1. allIngredients => ingredientsInPantry
2. "if has not" => if ingredient missing